solidity/contracts/system/VendingMachine.sol

```diff
@@ -74,11 +111,13 @@ contract VendingMachine is TBTCSystemAuthority{
 
         tbtcDepositToken.transferFrom(msg.sender, address(this), _tdtId);
 
-        // If the backing Deposit does not have a signer fee in escrow, mint it.
         Deposit deposit = Deposit(address(uint160(_tdtId)));
         uint256 signerFee = deposit.signerFee();
         uint256 depositValue = deposit.lotSizeTbtc();
 
+        require(getMintedSupply() + depositValue < getMaxSupply(), "Can't mint more than the max supply cap");
```

Comment:
Tbh it feels cleaner and clearer to have a `canMint(uint256 amount)` helper that does this check, and hook up this require and the `getAllowNewDeposits` check to that helper. Probably just bikeshedding at this point though.

Reply:
I'd rather extract the logic out of the `require` 🤷‍♂️

Reply:
`+` -> `.add()`